packages/network-controller/src/NetworkController.ts

```diff
@@ -446,6 +442,9 @@ export class NetworkController extends BaseControllerV2<
    * @param type - Human readable network name.
    */
   setProviderType(type: NetworkType) {
+    if (type === NetworkType.rpc) {
```

Comment:
Maybe we do this in another PR? It doesn't seem to be totally related to updating the network clients. I think it's worth doing a normalization step that ports over extra checks that the extension makes and anything else we need to do to bring this version closer to the extension, but maybe push that to another ticket?